### ISSUE TITLE
Record with a record in it gives `the Java file contained parse errors`

### ISSUE THREAD

**Issue by kajh** [revelc/impsort-maven-plugin#60]
**Describe the bug**

When running impsort-maven-plugin on a record with a record in it gives `the Java file contained parse errors` and `'record' cannot be used as an identifier as it is a keyword`.

I have added a demo project where this issue can be reproduced, see bellow.

**Versions (OS, Maven, Java, and others, as appropriate):**
 - Affected version(s) of this project: 1.6.2
 - OS: macOS 12.3.1
 - Others: 
    - Maven 3.8.5
    - Java 18.0.1  

**To Reproduce**
Steps to reproduce the behavior (or a link to an example repository that reproduces the problem):
1. Add the folloing file to a maven project where impsort-maven-plugin is configured
```
public record MyDTO(String myString, @JsonIgnore String name) {

    public PersonView getPerson() {
        return new PersonView(name);
    }

    record PersonView(String name) {
    }
}
```
2. Run `mvn package`

**Expected behavior**
No error messages

**Additional context**
I get the following error message:

 ```
$ mvn process-sources   
impsort-maven-plugin:1.6.2:sort (sort-imports) @ demo ---
(line 19,col 5) 'record' cannot be used as an identifier as it is a keyword.
[INFO] Processed 524 files in 00:00.653 (Already Sorted: 524, Needed Sorting: 0)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  3.786 s
[INFO] Finished at: 2022-05-24T20:59:34+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal net.revelc.code:impsort-maven-plugin:1.6.2:sort (sort-imports) on project demo: Error reading file MyDTO.java: file: MyDTO.java; reason: the Java file contained parse errors -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```
I have attached a demo project where this issue can be reprodused, please see [demo.zip](https://github.com/revelc/impsort-maven-plugin/files/8765734/demo.zip).

When I don't have `<exclude>**/MyDTO.java</exclude>` running `mvn process-sources` failed with the error message above.

When having `<exclude>**/MyDTO.java</exclude>`  running `mvn package` works fine, so the syntax in `MyDTO.java` seems to be ok

.

**Comment by ctubbsii** [comment]
This may be related to #54 , in that upstream javaparser may not yet support this.
If you can run with `mvn -e`, to get the full stack trace, that would help.

**Comment by ctubbsii** [comment]
You also reported a similar issue in #44

**Comment by kajh** [comment]
`mvn -e process-sources` gives the following output:

```
[INFO] --- impsort-maven-plugin:1.6.2:sort (sort-imports) @ demo ---
(line 5,col 1) Record Declarations are not supported.
[INFO] Processed 2 files in 00:00.195 (Already Sorted: 2, Needed Sorting: 0)
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.767 s
[INFO] Finished at: 2022-05-24T22:36:32+02:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal net.revelc.code:impsort-maven-plugin:1.6.2:sort (sort-imports) on project demo: Error reading file /Users/kajh/Downloads/demo/src/main/java/com/example/demo/MyDTO.java: file: /Users/kajh/Downloads/demo/src/main/java/com/example/demo/MyDTO.java; reason: the Java file contained parse errors -> [Help 1]
org.apache.maven.lifecycle.LifecycleExecutionException: Failed to execute goal net.revelc.code:impsort-maven-plugin:1.6.2:sort (sort-imports) on project demo: Error reading file /Users/kajh/Downloads/demo/src/main/java/com/example/demo/MyDTO.java
    at org.apache.maven.lifecycle.internal.MojoExecutor.doExecute (MojoExecutor.java:306)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:211)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:165)
    at org.apache.maven.lifecycle.internal.MojoExecutor.execute (MojoExecutor.java:157)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:121)
    at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject (LifecycleModuleBuilder.java:81)
    at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build (SingleThreadedBuilder.java:56)
    at org.apache.maven.lifecycle.internal.LifecycleStarter.execute (LifecycleStarter.java:127)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:294)
    at org.apache.maven.DefaultMaven.doExecute (DefaultMaven.java:192)
    at org.apache.maven.DefaultMaven.execute (DefaultMaven.java:105)
    at org.apache.maven.cli.MavenCli.execute (MavenCli.java:960)
    at org.apache.maven.cli.MavenCli.doMain (MavenCli.java:293)
    at org.apache.maven.cli.MavenCli.main (MavenCli.java:196)
    at jdk.internal.reflect.DirectMethodHandleAccessor.invoke (DirectMethodHandleAccessor.java:104)
    at java.lang.reflect.Method.invoke (Method.java:577)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced (Launcher.java:282)
    at org.codehaus.plexus.classworlds.launcher.Launcher.launch (Launcher.java:225)
    at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode (Launcher.java:406)
    at org.codehaus.plexus.classworlds.launcher.Launcher.main (Launcher.java:347)
Caused by: org.apache.maven.plugin.MojoFailureException: Error reading file /Users/kajh/Downloads/demo/src/main/java/com/example/demo/MyDTO.java
    at net.revelc.code.impsort.maven.plugin.AbstractImpSortMojo.fail (AbstractImpSortMojo.java:365)
    at net.revelc.code.impsort.maven.plugin.AbstractImpSortMojo.lambda$execute$3 (AbstractImpSortMojo.java:302)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:992)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:509)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (ForEachOps.java:150)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (ForEachOps.java:173)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:596)
    at java.util.stream.ReferencePipeline$7$1.accept (ReferencePipeline.java:276)
    at java.util.Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:992)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:509)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
    at java.util.stream.ReduceOps$ReduceTask.doLeaf (ReduceOps.java:960)
    at java.util.stream.ReduceOps$ReduceTask.doLeaf (ReduceOps.java:934)
    at java.util.stream.AbstractTask.compute (AbstractTask.java:327)
    at java.util.concurrent.CountedCompleter.exec (CountedCompleter.java:754)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec (ForkJoinPool.java:1182)
    at java.util.concurrent.ForkJoinPool.scan (ForkJoinPool.java:1655)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1622)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:165)
Caused by: net.revelc.code.impsort.ex.ImpSortException: file: /Users/kajh/Downloads/demo/src/main/java/com/example/demo/MyDTO.java; reason: the Java file contained parse errors
    at net.revelc.code.impsort.ImpSort.parseFile (ImpSort.java:133)
    at net.revelc.code.impsort.maven.plugin.AbstractImpSortMojo.lambda$execute$3 (AbstractImpSortMojo.java:287)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.accept (ForEachOps.java:183)
    at java.util.stream.ReferencePipeline$3$1.accept (ReferencePipeline.java:197)
    at java.util.Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:992)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:509)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
    at java.util.stream.ForEachOps$ForEachOp.evaluateSequential (ForEachOps.java:150)
    at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential (ForEachOps.java:173)
    at java.util.stream.AbstractPipeline.evaluate (AbstractPipeline.java:234)
    at java.util.stream.ReferencePipeline.forEach (ReferencePipeline.java:596)
    at java.util.stream.ReferencePipeline$7$1.accept (ReferencePipeline.java:276)
    at java.util.Spliterators$ArraySpliterator.forEachRemaining (Spliterators.java:992)
    at java.util.stream.AbstractPipeline.copyInto (AbstractPipeline.java:509)
    at java.util.stream.AbstractPipeline.wrapAndCopyInto (AbstractPipeline.java:499)
    at java.util.stream.ReduceOps$ReduceTask.doLeaf (ReduceOps.java:960)
    at java.util.stream.ReduceOps$ReduceTask.doLeaf (ReduceOps.java:934)
    at java.util.stream.AbstractTask.compute (AbstractTask.java:327)
    at java.util.concurrent.CountedCompleter.exec (CountedCompleter.java:754)
    at java.util.concurrent.ForkJoinTask.doExec (ForkJoinTask.java:373)
    at java.util.concurrent.ForkJoinPool$WorkQueue.topLevelExec (ForkJoinPool.java:1182)
    at java.util.concurrent.ForkJoinPool.scan (ForkJoinPool.java:1655)
    at java.util.concurrent.ForkJoinPool.runWorker (ForkJoinPool.java:1622)
    at java.util.concurrent.ForkJoinWorkerThread.run (ForkJoinWorkerThread.java:165)
[ERROR] 
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/MojoFailureException
```

**Comment by kajh** [comment]
Regarding #44, I see that the issue in the parser which is refered to there is closed, ref. https://github.com/javaparser/javaparser/issues/3261.

**Comment by ctubbsii** [comment]
Should be fixed in 146a0f1617747d550ad9a66e3437c640c29f9660
However, you will need to set the compiler compliance level to 17, either directly in the plugin configuration, or by setting `maven.compiler.release` to 17 in your properties.
